_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -105,6 +105,12 @@ public static function register_endpoints() {
 			'callback' => __CLASS__ . '::jetpack_connection_status',
 		) );
 
+		// Test current connection status of Jetpack
+		register_rest_route( 'jetpack/v4', '/connection/test', array(
+			'methods' => WP_REST_Server::READABLE,
+			'callback' => __CLASS__ . '::jetpack_connection_test',
```

Comment:
Should we have a `permission_callback` as well?

Reply:
Yes thank you. I used our `jetpack_admin_page` capability. It defaults to `read` which is the same caps that the wpcom endpoint requires.

Reply:
Anyone with a user account on the Jetpack site (not necessarily an account that's connected to WP.com) can `read`. We use `read` for the admin page since we show the dashboard to everyone with an account on the site (so that they can connect to WP.com).

Maybe lock it down to `::manage_modules_permission_check()`? I don't have any specific objections to `jetpack_admin_page` - just not sure if it's necessary.

Reply:
I'll trust whatever decision you make :)

Reply:
IDK - we'd only lose the ability to use this endpoint if we're working on something that requires us to be logged in as a low-capability user. Using the Jetpack capabilities only provides support for those that modify them, like VIP or so. 

I'm fine to use `manage_modules` check for now. Covers the vast majority of our use-case currently. If we end up wanting to use this endpoint for more things than just sanity checks during sandboxing in the future, we can revisit.

Reply:
Done in 6adc890!

Reply:
I had an idea we could implement as a followup: we could set the permission check to `jetpack_admin_page` if it's sandboxed and `manage_modules_permission_check` if it's not.